mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/bundle/ops/XGBoostPredictorClassificationOp.scala

```diff
@@ -0,0 +1,49 @@
+package ml.combust.mleap.xgboost.runtime.bundle.ops
+
+import java.nio.file.Files
+
+import biz.k11i.xgboost.Predictor
+import ml.combust.bundle.BundleContext
+import ml.combust.bundle.dsl.{Model, Value}
+import ml.combust.bundle.op.OpModel
+import ml.combust.mleap.bundle.ops.MleapOp
+import ml.combust.mleap.runtime.MleapContext
+import ml.combust.mleap.xgboost.runtime._
+
+
+/**
+  * Created by hollinwilkins on 9/16/17.
+  */
+class XGBoostPredictorClassificationOp extends MleapOp[XGBoostPredictorClassification, XGBoostPredictorClassificationModel] {
+
+  override val Model: OpModel[MleapContext, XGBoostPredictorClassificationModel] = new OpModel[MleapContext, XGBoostPredictorClassificationModel] {
+  override val klazz: Class[XGBoostPredictorClassificationModel] = classOf[XGBoostPredictorClassificationModel]
+
+  override def opName: String = "xgboost.classifier"
+
+  @throws[RuntimeException]
+  override def store(model: Model, obj: XGBoostPredictorClassificationModel)
+                    (implicit context: BundleContext[MleapContext]): Model =
+    throw new RuntimeException("The XGBoostPredictor implementation does not support storing the model.")
```

Comment:
Is it because the underlying library doesn't support this?

Reply:
Yes correct. The library's only purpose is to load an xjboost4j model runtime to predict/run/infer with higher performance than the officail xgboost4j loader.

They achieve this by re-implementing the runtime code in Java which is faster than going through C++  (which is what xgboost4j does). Going to C++ is fine in general, but doesn't really scale on "big data" traffic, so I think it's good to provide both options in MLeap.

Some of the benchmarks are on their main page: https://github.com/komiya-atsushi/xgboost-predictor-java
but one of our teammates has also run independent benchmarks here: https://github.com/combust/mleap/issues/631

Reply:
makes sense, thank you!